pandas/core/indexes/datetimelike.py

```diff
@@ -70,15 +71,29 @@ class DatetimeIndexOpsMixin(ExtensionOpsMixin):
     _maybe_mask_results = ea_passthrough("_maybe_mask_results")
     __iter__ = ea_passthrough("__iter__")
 
+    @property
+    def _eadata(self):
+        return self._data
+
+    @property
+    def freq(self):
+        # Can't simply use delegate_names since our base class is defining
+        # freq
+        return self._data.freq
+
+    @freq.setter
+    def freq(self, value):
+        self._data.freq = value
+
     @property
     def freqstr(self):
-        return self._eadata.freqstr
+        return self._data.freqstr
```

Comment:
Do we want to do these kinds of changes here, or in a followup? I've been doing them sporadically, but I think a followup would be better to keep this diff smaller.

Reply:
agreed